src/Hangfire.Core/Dashboard/Content/js/hangfire.js

```diff
@@ -457,8 +459,177 @@
+                    
+                    url = document.location.search;
+
+                    var filterValueString = $("#filterValueString").val();
+                    var filterStartDate = $("#startDate").datepick({ dateFormat: 'dd-mm-yyyy' }).val();
+                    var filterEndDate = $("#endDate").datepick({ dateFormat: 'dd-mm-yyyy' }).val();
+
+                    var addedFilterString = prepFilterStringParameter(filterValueString);    
+                    
+                    var addedDateFilterStrings = prepFilterDateParameters(filterStartDate, filterEndDate);
+                    
+                    
+                    if (addedDateFilterStrings || addedFilterString) {                                  
+                        redirectToFirstPage();                        
+                    }                    
+                    document.location.search = url;
+                                      
+                })
+
+
+                var prepFilterStringParameter = function (filterString) {
+                    
+                    var result = false;
+                    if (url == '' && filterString != '') {                        
+                        url = '?' + "filterString=" + filterString;                       
+                        result = true;
+                    }
+                    else {
+                        var parameters = url.substr(1).split('&');
+                        var existingParameters = false;
+                        var x;
+                        var i = 0;
+                        for(i=0;i < parameters.length;i++){
+                            x = parameters[i].split('=');
+                            if (x[0] == "filterString" && filterString != '' ) {
+                                x[1] = filterString;
+                                parameters[i] = x.join('=');                                
+                                result = true;
+                                break;
+                            }
+                            else if (x[0] == "filterString" && filterString == '') {
+                                parameters.splice(i, 1);                                
+                                result = false;
+                                break;
+                            }
+                        }
+
+                        if ( i >= parameters.length && filterString != '' ) {
+                            parameters[parameters.length] = ["filterString", filterString].join('=');  
+                            result = true;
+                        }
+                        
+                        url = '?' + parameters.join('&');
+                        
+                    }                   
+                    return result;                    
+                }
+
+                var prepFilterDateParameters = function (startDate, endDate) {
+
+                    var sDate = startDate.split('/');
```

Comment:
Are joined variable declarations common in this codebase? If not is this a convention that everyone agrees with?